### PR TITLE
Stepper framework: Add import flow

### PR DESCRIPTION
**Pull request by bogiii**
#### Changes proposed in this Pull Request

This PR contains the first part of the effort introducing the whole import onboarding experience to the stepper framework.

- 🟢 `start/importer/capture` (signup) -> `setup/import` (stepper)
- ⚪️ `start/from/importing/{importer}` (signup) -> `setup/importer/{importer}`

* Registered an `import` flow inside a new non-linear stepper framework.
* Adjusted the stepper framework allowing consumption of the main redux store since the import module depends on it.
* Setup import flow to consume existing import module screens from the linear signup framework.

**Next steps** (not covered by this PR):
- Move the place where the import module lives. From the signup steps to `calypso/blocks`
- Register an importer flow equivalently with `/start/from/{importer}`

#### Testing instructions

* Go to the `setup/intent?siteSlug={SLUG}`
* Select `Import your site content` option
* Enter any URL you want to import from
* Check if it works in the same way as it is the case when you start from `/start/setup-site/intent?siteSlug={SLUG}`
* Check navigation `Back` and `I don't have a site address` header's navigation links
* In other words, it should be the same as it is on the `/start` signup flow

Related to #62238

Additonal information: pcmemI-u6-p2

### CODE REVIEW

**Review by vishnugopal — APPROVED** [review]
Code looks great and most functionality works fine. Found a possible tiny bug:

https://user-images.githubusercontent.com/533/164383525-6ef9e722-9a36-4c57-ae0e-c15de774600d.mp4

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+	/**
+	 ↓ Fields
+	 */
+	const siteSlug = useSiteSlugParam();
+	const site = useSite();
+	const currentRoute = useCurrentRoute();
+	const isAtomicSite = useSelector( ( state ) =>
+		isAtomicSiteSelector( state, site?.ID as number )
+	);
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: Trying to avoid Calypso state when possible:

```suggestion
	const isAtomicSite = site?.options?.is_automated_transfer;
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import { getUrlData } from 'calypso/state/imports/url-analyzer/selectors';
+import isAtomicSiteSelector from 'calypso/state/selectors/is-site-automated-transfer';
+import { getFinalImporterUrl } from './helper';
+import { redirect, removeTrailingSlash } from './util';
+import type { Step } from '../../types';
+import './style.scss';
+
+const ImportStep: Step = function ImportStep( props ) {
+	const { __ } = useI18n();
+	const BASE_ROUTE = 'import';
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: I'd put this outside the component, it never changes.

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+	function shouldHideSkipBtn() {
+		switch ( currentRoute ) {
+			case 'import':
+				return false;
+
+			default:
+				return true;
+		}
+	}
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]:
```suggestion
	const shouldHideSkipBtn = currentRoute !== 'import'
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+	/**
+	 ↓ Methods
+	 */
+	const goToStep: GoToStep = function ( stepName, stepSectionName ) {
+		const routes = [ BASE_ROUTE, stepName, stepSectionName ];
+		const stepPath = removeTrailingSlash( routes.join( '/' ) ) as StepPath;
+
+		navigation.goToStep?.( stepPath );
+	};
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: I think we only need this because we're creating a tiny router below.

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+				) }
+
+				{ currentRoute === 'import/ready/wpcom' && (
+					<ReadyAlreadyOnWPCOMStep
+						urlData={ urlData }
+						goToStep={ goToStep }
+						recordTracksEvent={ recordTracksEvent }
+					/>
+				) }
+			</div>
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: I'm confused as to why we're hacking our own router here. Would it be hard to make these individual steps? Maybe with a common wrapper ImportWrapper or something if they share some logic? They're already neatly compartmentalized components. 

When we make the individual steps we get 1) free routing 2) free reusability 3) smaller files that are easier to understand.

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+		);
+	}
+
+	return (
+		<>
+			<DocumentHead title={ __( 'Import your site content' ) } />
+
+			<StepContainer
+				stepName={ 'import-step' }
+				hideSkip={ shouldHideSkipBtn() }
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]:
```suggestion
				hideSkip={ shouldHideSkipBtn }
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/index.ts [hunk 7]
```diff
@@ -19,6 +20,12 @@ export type StepPath =
 	| 'bloggerStartingPoint'
 	| 'storeFeatures'
 	| 'designSetup'
+	| 'import'
+	| 'import/list'
+	| 'import/ready'
+	| 'import/ready/not'
+	| 'import/ready/wpcom'
+	| 'import/ready/preview'
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/index.ts]: Unless it's a deal breaker for you, I'd aim to maintain the flatness of the steps. As in change `import/list` to `import-list`, `import-ready`, etc.. 

I entirely agree these steps are part of `import`, but it would be nice to establish a sense of flat steps-repository; every step is simple a nice component connected to the store. This makes every step directly reusable in other flows when we inevitably create them.

```suggestion
	| 'import-list'
	| 'import-ready'
	| 'import-ready-not'
	| 'import-ready-wpcom'
	| 'import-ready-preview'
```

Then you'll be free from navigation magic as well, you won't need to remove trailing slashes.

**Review by alshakero — CHANGES_REQUESTED** [review]
Super excited for this one! thanks a lot!

Left some comments, mostly optional but would be greatly appreciated if addressed. These are the first flows to join stepper, and I'm trying to create good examples for the dozen flows that will follow; most people will look at the code and copy.

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx (hunk 3 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: Agree! I've added a shorthanded version.

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx (hunk 2 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: 👍

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx (hunk 5 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/import/index.tsx]: I did not pay extra attention, a similar setup already exists in the linear framework, so I just took the same logic.

Agree with you, and I'll try to separate it into individual steps using existing routing logic.

Thanks!

File: client/landing/stepper/declarative-flow/internals/steps-repository/index.ts (hunk 7 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/index.ts]: My idea was to reuse the same route structure as we have on `/start/import/{STEP}/{SECTION}`, but it's not a deal-breaker to update routes as you proposed (it's always good to be consistent).

**Comment by bogiii** [comment]
> Code looks great and most functionality works fine. Found a possible tiny bug:
> 
>  CleanShot.2022-04-21.at.11.20.43.mp4

Thanks for discovering! It has been fixed.

cc @vishnugopal

File: client/landing/stepper/declarative-flow/internals/steps-repository/index.ts (hunk 7 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/index.ts]: It has been updated. I've changed slashes routes to the camel case pattern to follow existing route patterns.

`import/ready/preview` -> `importReadyPreview`

**Comment by bogiii** [comment]
@alshakero, could you please take one more look to confirm?

I would like to merge it soon.

File: client/landing/stepper/declarative-flow/internals/steps-repository/import/helper.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	// Escape WordPress, has two sub-flows "Import everything" and "Content only"
+	// firstly show import type chooser screen and then decide about importer url
+	if ( isAtomicSite && platform !== 'wordpress' ) {
+		importerUrl = getWpOrgImporterUrl( targetSlug, platform );
+	} else if (
+		( platform === 'blogger' && isEnabled( 'onboarding/import-from-blogger' ) ) ||
+		( platform === 'medium' && isEnabled( 'onboarding/import-from-medium' ) ) ||
+		( platform === 'squarespace' && isEnabled( 'onboarding/import-from-squarespace' ) ) ||
+		( platform === 'wix' && isEnabled( 'onboarding/import-from-wix' ) ) ||
+		( platform === 'wordpress' && isEnabled( 'onboarding/import-from-wordpress' ) )
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/import/helper.ts]: 100% optional:

```suggestion
		['blogger', 'medium', 'squarespace', 'wix', 'wordpress'].some(platform => isEnabled( `onboarding/import-from-${platform}` ) )
```

**Review by alshakero — APPROVED** [review]
Thank you so so much for this! Looks great now!
